neuralmonkey/learning_utils.py

```diff
@@ -156,48 +176,55 @@ def training_loop(tf_manager: TensorFlowManager,
 
                 if step % validation_period == validation_period - 1:
-                    val_results, val_outputs = run_on_dataset(
-                        tf_manager, runners, val_dataset,
-                        postprocess, write_out=False,
-                        batch_size=runners_batch_size)
-                    # ensure val outputs are iterable more than once
-                    val_outputs = {k: list(v) for k, v in val_outputs.items()}
-                    val_evaluation = evaluation(
-                        evaluators, val_dataset, runners, val_results,
-                        val_outputs)
-
-                    this_score = val_evaluation[main_metric]
-                    tf_manager.validation_hook(this_score, epoch_n, batch_n)
-
-                    log("Validation (epoch {}, batch number {}):"
-                        .format(epoch_n, batch_n), color='blue')
-
-                    _log_continuous_evaluation(tb_writer, tf_manager,
-                                               main_metric,
-                                               val_evaluation,
-                                               seen_instances, epoch_n,
-                                               epochs,
-                                               val_results, train=False)
-
-                    if this_score == tf_manager.best_score:
-                        best_score_str = colored(
-                            "{:.4g}".format(tf_manager.best_score),
-                            attrs=['bold'])
-                    else:
-                        best_score_str = "{:.4g}".format(tf_manager.best_score)
-
-                    log("best {} on validation: {} (in epoch {}, "
-                        "after batch number {})"
-                        .format(main_metric, best_score_str,
-                                tf_manager.best_score_epoch,
-                                tf_manager.best_score_batch),
-                        color='blue')
-
-                    log_print("")
-                    _print_examples(val_dataset, val_outputs,
-                                    val_preview_input_series,
-                                    val_preview_output_series,
-                                    val_preview_num_examples)
+                    for val_id, valset in enumerate(val_datasets):
+                        val_results, val_outputs = run_on_dataset(
+                            tf_manager, runners, valset,
+                            postprocess, write_out=False,
+                            batch_size=runners_batch_size)
+                        # ensure val outputs are iterable more than once
+                        val_outputs = {k: list(v)
+                                       for k, v in val_outputs.items()}
+                        val_evaluation = evaluation(
+                            evaluators, valset, runners,
+                            val_results,
+                            val_outputs)
```

Comment:
nevejde se to na jednu řádku? Jinak teda aspoň na dvě, tohle prosím posuň k `val_results`